keeper-gradle-plugin/src/test/kotlin/com/slack/keeper/KeeperFunctionalTest.kt

```diff
@@ -262,40 +269,23 @@ private fun String.prefixIfNot(prefix: String) =
     if (this.startsWith(prefix)) this else "$prefix$this"
 
 @Language("PROGUARD")
-private val EXPECTED_PRINT_RULES_RULES = """
-  -keep class com.slack.keeper.sample.TestOnlyClass {
-    public static void testOnlyMethod();
-  }
-  -keep class com.slack.keeper.sample.TestOnlyKotlinClass {
-    public void testOnlyMethod();
-    com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;
-  }
-  -keeppackagenames
-""".trimIndent()
+private val EXPECTED_TRACE_REFERENCES_CONFIG: Map<String, List<String>?> = mapOf(
+  "-keep class com.slack.keeper.sample.TestOnlyClass" to listOf(
+    "public static void testOnlyMethod();"
+  ),
+  "-keep class com.slack.keeper.sample.TestOnlyKotlinClass" to listOf(
+    "public void testOnlyMethod();",
+    "com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;"
+  )
+)
 
 @Language("PROGUARD")
-private val EXPECTED_TRACE_REFERENCES_CONFIG = """
-  -keep class com.slack.keeper.sample.TestOnlyClass {
-      public static void testOnlyMethod();
-  }
-  
-  -keep class com.slack.keeper.sample.TestOnlyKotlinClass {
-      com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;
-      public void testOnlyMethod();
-  }
-""".trimIndent()
+private val EXPECTED_PRINT_RULES_CONFIG =
+  EXPECTED_TRACE_REFERENCES_CONFIG + listOf("-keeppackagenames" to null)
 
-@Language("PROGUARD")
-private val EXPECTED_PROGUARD_CONFIG = """
-  -keep class com.slack.keeper.sample.TestOnlyClass {
-      public static void testOnlyMethod();
-  }
-  
-  -keep class com.slack.keeper.sample.TestOnlyKotlinClass {
-      com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;
-      public void testOnlyMethod();
-  }
-""".trimIndent()
+private fun indentRules(header: String, content: List<String>?) =
```

Comment:
Can you doc this a bit? I didn't follow what the methodology here was at first

Reply:
It's allowing to generate a proguard rule block on demand.
But I still need it because depending on if R8 is enabled or not, rules output is different as was defined in https://github.com/slackhq/keeper/blob/main/keeper-gradle-plugin/src/test/kotlin/com/slack/keeper/KeeperFunctionalTest.kt#L266-L268
```
  -keep class com.slack.keeper.sample.TestOnlyKotlinClass {
    public void testOnlyMethod();
    com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;
  }
```
and https://github.com/slackhq/keeper/blob/main/keeper-gradle-plugin/src/test/kotlin/com/slack/keeper/KeeperFunctionalTest.kt#L278-L280
```
    -keep class com.slack.keeper.sample.TestOnlyKotlinClass {
        com.slack.keeper.sample.TestOnlyKotlinClass INSTANCE;
        public void testOnlyMethod();
    }
```

Also just for reference, at the beginning I was providing the indent level (2 or 4) but later on I preferred to normalize the assertion input with `replace("    ", "  ")`, but as you noticed it's still helpful for the content ordering 👍